core/lib/FrameNavigationsObserver.ts

```diff
@@ -162,14 +166,11 @@ export default class FrameNavigationsObserver {
     for (const state of this.navigations.top.stateChanges.keys()) {
       // don't resolve states for redirected
       if (state === LocationStatus.HttpRedirected) continue;
-      let recordedStatus = PipelineStatus[state as IPipelineStatus];
-
-      // use painting stable "order of pipeline" if the passed in state is load
+      let pipelineStatus = PipelineStatus[state as IPipelineStatus];
       if (state === LoadStatus.Load) {
-        recordedStatus = PipelineStatus.PaintingStable;
+        pipelineStatus = PipelineStatus.AllContentLoaded;
```

Comment:
Why the switch from PaintingStable to AllContentLoaded?

Reply:
Painting Stable actually gets handled separately right above this snippet. This is assigning a pipeline status to the "load" event that gets stored in the load statuses.